lapis/utilities/cache_algorithm_implementations.py

```diff
@@ -0,0 +1,5 @@
+def cache_all():
```

Comment:
It may be worth combing these with the ``cleanup`` algorithms. For the most parts, cache algorithms do not separate "add" and "remove"; new content is considered (potentially with regard to old one) and if there is not enough space *some* item is discarded - be it a new or existing one.

Consider stateful objects, of which each cache owns one:
```
class CacheAlgorithm:
    # if state is needed, add it in __init__
    def consider(self, *candidates: File) -> Set[File]:
           """
           Consider adding new file and return a list of files to evict

           If a new file is not to be added, it must be in the eviction set.
           """
```